From: Horst Beepmanh <>
Subject: Makefile fixes

---
diff --git a/Drivers/Drivers.make b/Drivers/Drivers.make
--- a/Drivers/Drivers.make
+++ b/Drivers/Drivers.make
@@ -22,9 +22,6 @@ BackLightDir	=	{DriverDir}BackLight:
 
 #include {SonyDir}Sony.make
 
-#include {SerialDir}Serial.make
-#include {SerialDMADir}SerialDMA.make
-
 #include {NewAgeDir}NewAge.make
 
 #include {IOPDir}IOP.make
diff --git a/Make/DBLite.make b/Make/DBLite.make
--- a/Make/DBLite.make
+++ b/Make/DBLite.make
@@ -254,7 +254,7 @@ Clean							ƒ
 .cp.o							ƒ	.c
 	{CPlus} {StdCPOpts} -o {Targ} {DepDir}{Default}.cp
 
-#include {ToolSrcDir}Tools.make
+#include {DriverDir}Drivers.make
 
 #include {MakeDir}MainCode.make
 
@@ -262,7 +262,5 @@ Clean							ƒ
 
 #include {DeclDir}VSCDeclData:VSCDeclData.make
  
-#include {DriverDir}Drivers.make
-
 #include {ResourceDir}Resources.make
 
diff --git a/Make/LC930.make b/Make/LC930.make
--- a/Make/LC930.make
+++ b/Make/LC930.make
@@ -236,12 +236,10 @@ Clean							ƒ
 	{CPlus} {StdCPOpts} -o {Targ} {DepDir}{Default}.cp
 
 
-#include {ToolSrcDir}Tools.make
+#include {DriverDir}Drivers.make
 
 #include {MakeDir}MainCode.make
 
 #include {DeclDir}DeclData.make
 
-#include {DriverDir}Drivers.make
-
 #include {ResourceDir}Resources.make
diff --git a/Make/RISC.make b/Make/RISC.make
--- a/Make/RISC.make
+++ b/Make/RISC.make
@@ -379,12 +379,10 @@ Clean							ƒ
 .cp.o							ƒ	.c
 	{CPlus} {StdCPOpts} -o {Targ} {DepDir}{Default}.cp
 
-#include {ToolSrcDir}Tools.make
+#include {DriverDir}Drivers.make
 
 #include {MakeDir}MainCode.make
 
 #include {DeclDir}DeclData.make
 
-#include {DriverDir}Drivers.make
-
 #include {ResourceDir}Resources.make
diff --git a/Make/Universal.make b/Make/Universal.make
--- a/Make/Universal.make
+++ b/Make/Universal.make
@@ -367,12 +367,10 @@ Clean							ƒ
 	{Pascal} {StdPOpts} -o {Targ} {DepDir}{Default}.p
 
 
-#include {ToolSrcDir}Tools.make
+#include {DriverDir}Drivers.make
 
 #include {MakeDir}MainCode.make
 
 #include {DeclDir}DeclData.make
 
-#include {DriverDir}Drivers.make
-
 #include {ResourceDir}Resources.make
diff --git a/OS/MemoryMgr/MemoryMgr.make b/OS/MemoryMgr/MemoryMgr.make
--- a/OS/MemoryMgr/MemoryMgr.make
+++ b/OS/MemoryMgr/MemoryMgr.make
@@ -17,15 +17,15 @@
 
 FigmentDir = {MemoryMgrDir}FigmentSources:
 
-#include {FigmentDir}Figment.Make
-
-
 MemoryMgrObjs 					=	"{ObjDir}MemoryMgr.a.o"							∂
 									"{ObjDir}MemoryMgrExtensions.a.o"				∂
 									"{ObjDir}MemoryMgrInternal.a.o"					∂
 									"{ObjDir}BlockMove.a.o"
 
 
+#include {FigmentDir}Figment.Make
+
+
 "{LibDir}MemoryMgr.lib"			ƒ	"{LibDir}Figment.lib" {MemoryMgrObjs}
 	If "{FeatureSet}" =~ /≈'hasNewHeapMgr=True'≈/
 	 	Lib {StdLibOpts} -o "{Targ}" "{LibDir}Figment.lib" {MemoryMgrObjs}
diff --git a/OS/OS.make b/OS/OS.make
--- a/OS/OS.make
+++ b/OS/OS.make
@@ -67,14 +67,12 @@ RealTimeDir		=	{OSDir}RealTime:
 #include {ADBDir}ADBMgr.make
 #include {TimeMgrDir}TimeMgr.make
 #include {MemoryMgrDir}MemoryMgr.make
-#include {BeepDir}Beep.make
 #include {FPEmulationDir}FPEmulation.make
 #include {GestaltDir}Gestalt.make
 #include {KeyboardDir}Keyboard.make
 #include {NetBootDir}NetBoot.make
 #include {VDigDir}VDig.make
 #include {I2CDir}I2C.make
-#include {RealTimeDir}RealTime.make
 
 
 OSObjs	= 							"{ObjDir}clock.a.o"								∂
diff --git a/Toolbox/Toolbox.make b/Toolbox/Toolbox.make
new file mode 100644
--- /dev/null
+++ b/Toolbox/Toolbox.make
@@ -0,0 +1,110 @@
+# This makefile was missing. Here is a rudimentary reconstruction.
+
+AliasMgrDir 				= "{ToolBoxDir}AliasMgr:"
+AppleEventDir				= "{ToolBoxDir}AppleEventMgr:"
+ColorPickerDir				= "{ToolBoxDir}ColorPicker:"
+CommToolboxDir				= "{ToolBoxDir}CommToolbox:"
+ComponentMgrDir 			= "{ToolBoxDir}ComponentMgr:"
+ControlMgrDir 				= "{ToolBoxDir}ControlMgr:"
+DataAccessDir 				= "{ToolBoxDir}DataAccessMgr:"
+DialogDir 					= "{ToolBoxDir}DialogMgr:"
+DictionaryMgrDir			= "{ToolBoxDir}DictionaryMgr:"
+DisplayMgrDir 				= "{ToolBoxDir}DisplayMgr:"
+EditionMgrDir				= "{ToolBoxDir}DataPubsMgr:"
+ExpansionBusMgrDir 			= "{ToolBoxDir}ExpansionBusMgr:"
+FontMgrDir 					= "{ToolBoxDir}FontMgr:"
+HelpMgrDir					= "{ToolBoxDir}HelpMgr:"
+IconUtilsDir				= "{ToolBoxDir}IconUtils:"
+InSaneDir 					= "{ToolBoxDir}InSANE:"
+ListMgrDir					= "{ToolBoxDir}ListMgr:"
+MenuMgrDir 					= "{ToolBoxDir}MenuMgr:"
+NotificationDir 			= "{ToolBoxDir}NotificationMgr:"
+PrintingDir					= "{ToolBoxDir}Printing:"
+ResourceMgrDir 				= "{ToolBoxDir}ResourceMgr:"
+SANEDir 					= "{ToolBoxDir}SANE:"
+ScrapMgrDir					= "{ToolBoxDir}ScrapMgr:"
+ScriptMgrDir 				= "{ToolBoxDir}ScriptMgr:"
+StandardFileDir 			= "{ToolBoxDir}StandardFile:"
+TextServicesDir				= "{ToolBoxDir}TextServicesMgr:"
+ToolboxEventDir 			= "{ToolBoxDir}ToolboxEventMgr:"
+WindowMgrDir 				= "{ToolBoxDir}WindowMgr:"
+
+
+#include "{AliasMgrDir}AliasMgr.make"
+#include "{AppleEventDir}AppleEventMgr.make"
+#include "{ColorPickerDir}ColorPicker.make"
+#include "{CommToolboxDir}CommToolbox.make"
+#include "{ComponentMgrDir}ComponentMgr.make"
+#include "{ControlMgrDir}ControlMgr.make"
+#include "{DataAccessDir}DataAccessMgr.make"
+#include "{DialogDir}DialogMgr.make"
+#include "{DictionaryMgrDir}DictionaryMgr.make"
+#include "{DisplayMgrDir}DisplayMgr.make"
+#include "{EditionMgrDir}EditionMgr.make"
+#include "{ExpansionBusMgrDir}ExpansionBusMgr.make"
+#include "{FontMgrDir}FontMgr.make"
+#include "{HelpMgrDir}HelpMgr.make"
+#include "{IconUtilsDir}IconUtils.make"
+#include "{InSaneDir}InSane.make"
+#include "{ListMgrDir}ListMgr.make"
+#include "{MenuMgrDir}MenuMgr.make"
+#include "{NotificationDir}NotificationMgr.make"
+#include "{ResourceMgrDir}ResourceMgr.make"
+#include "{SANEDir}SANE.make"
+#include "{ScriptMgrDir}ScriptMgr.make"
+#include "{StandardFileDir}StandardFile.make"
+#include "{TextServicesDir}TextServicesMgr.make"
+#include "{ToolboxEventDir}ToolboxEventMgr.make"
+#include "{WindowMgrDir}WindowMgr.make"
+
+
+ToolboxObjs	= 						"{ObjDir}PackageMgr.a.o"						∂
+									"{ObjDir}SegmentLoader.a.o"						∂
+									"{ObjDir}ShutDownMgr.a.o"						∂
+									"{ObjDir}Munger.a.o"							∂
+									"{ObjDir}DeskMgr.a.o"							∂
+									"{ObjDir}GetMgr.a.o"							∂
+									"{ObjDir}TextEdit.a.o"							∂
+									"{ObjDir}ScrapMgr.a.o"							∂
+									"{ObjDir}PrintGlue.a.o"							∂
+									"{ObjDir}SexyDate.a.o"							∂
+
+
+# Just guessing what the binaries should be called:
+# *.lib for multi-file managers, *.a.o for single-file managers
+
+"{LibDir}Toolbox.lib"				ƒ	{ToolboxObjs}
+	Lib {StdLibOpts}	{ToolboxObjs} -o "{Targ}"
+
+"{ObjDir}PackageMgr.a.o"			ƒ	"{ToolboxDir}PackageMgr.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}PackageMgr.a"
+
+"{ObjDir}SegmentLoader.a.o"			ƒ	"{ToolboxDir}SegmentLoader:SegmentLoader.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}SegmentLoader:SegmentLoader.a"
+
+"{ObjDir}ShutDownMgr.a.o"			ƒ	"{ToolboxDir}ShutDownMgr:ShutDownMgr.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}ShutDownMgr:ShutDownMgr.a"
+
+"{ObjDir}Munger.a.o"				ƒ	"{ToolboxDir}Munger:Munger.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}Munger:Munger.a"
+
+"{ObjDir}DeskMgr.a.o"				ƒ	"{ToolboxDir}DeskMgr:DeskMgr.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}DeskMgr:DeskMgr.a"
+
+"{ObjDir}GetMgr.a.o"				ƒ	"{ToolboxDir}GetMgr:GetMgr.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}GetMgr:GetMgr.a"
+
+"{ObjDir}TextEdit.a.o"				ƒ	"{ToolboxDir}TextEdit:TextEdit.a" "{ToolboxDir}TextEdit:IncludeTextEdit.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}TextEdit:IncludeTextEdit.a"
+
+"{ObjDir}SexyDate.a.o"				ƒ	"{ToolboxDir}SexyDate.a"
+	Asm {StdAOpts} -o "{Targ}" "{ToolboxDir}SexyDate.a"
+
+"{ObjDir}ScrapMgr.a.o"				ƒ	"{ObjDir}StandardEqu.d"						∂
+										"{ScrapMgrDir}ScrapMgr.a"
+	Asm {StdAOpts} -o "{Targ}" "{ScrapMgrDir}ScrapMgr.a"
+
+"{ObjDir}PrintGlue.a.o"				ƒ	"{ObjDir}StandardEqu.d"						∂
+										"{AIncludes}PrPrivate.a"					∂
+										"{PrintingDir}PrintGlue.a"
+	Asm {StdAOpts} -o "{Targ}" "{PrintingDir}PrintGlue.a"
diff --git a/Toolbox/Toolbox.make.idump b/Toolbox/Toolbox.make.idump
new file mode 100644
--- /dev/null
+++ b/Toolbox/Toolbox.make.idump
@@ -0,0 +1 @@
+TEXTMPS 
\ No newline at end of file
