From: Horst Beepmanh <>
Subject: Amphibian DNA

---
--- /dev/null
+++ b/AmphibianDNA/ATAMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/AliasMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/BCScreen.a.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/BlueBoxPrintHook.c.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/ComponentMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/DeviceDispatch.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/DeviceMgr.a.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/DisplayMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/ExpansionBusMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/Gestalt.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/HFS.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/InterruptHandlers.a.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/MemoryMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/MenuMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/MoreCQD.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/NVRAMPrimitives.c.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/OpenFirmware.c.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/PPCDebuggerNub.a.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/PPCDebuggerNub.c.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/PowerMgr.a.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/PowerMgr.c.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/SCSI.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/ScriptMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/ShutDownMgr.c.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/StartLibGlue.c.o.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file





--- /dev/null
+++ b/AmphibianDNA/WindowMgr.lib.idump
@@ -0,0 +1 @@
+OBJ MPS 
\ No newline at end of file
-- 
2.18.0

